fmriprep/workflows/bold/resampling.py

```diff
@@ -158,7 +200,383 @@ def select_target(subject_id, space):
         name="outputnode",
     )
 
-    # fmt:off
+    # 0, 1 = wm; 2, 3 = pial; 6, 7 = mid
+    # note that order of lh / rh within each surf type is not guaranteed due to use
+    # of unsorted glob by FreeSurferSource prior, but we can do a sort with
+    # _sorted_by_basename to ensure consistent ordering
+    select_wm = pe.Node(
+        niu.Select(index=[0, 1]),
+        name="select_wm",
+        mem_gb=DEFAULT_MEMORY_MIN_GB,
+    )
+
+    select_pial = pe.Node(
+        niu.Select(index=[2, 3]),
+        name="select_pial",
+        mem_gb=DEFAULT_MEMORY_MIN_GB,
+    )
+
+    select_midthick = pe.Node(
+        niu.Select(index=[6, 7]),
+        name="select_midthick",
+        mem_gb=DEFAULT_MEMORY_MIN_GB,
+    )
+
+    create_wm_distvol = pe.MapNode(
+        CreateSignedDistanceVolume(),
+        iterfield=["surface"],
```

Comment:
This should really be its own workflow at this point. In this workflow, we could then have a small adaptation:

```Python
if project_goodvoxels:
    ribbon_mask_wf = init_ribbon_mask_wf()
    apply_ribbon_mask = pe.Node(
        fsl.ApplyMask(),
        name_source=['in_file'],
        name="apply_goodvoxels_ribbon_mask",
        mem_gb=mem_gb * 3,
    )

    wf.connect([
        (inputnode, ribbon_mask_wf, [('anat_giftis', 'anat_giftis'),
                                     ('t1w_mask', 't1w_mask')]),
        (rename_src, apply_ribbon_mask, [('out_file', 'in_file')]),
        (ribbon_mask_wf, apply_ribbon_mask, [('mask_file', 'mask_file')]),
        (apply_ribbon_mask, sampler, [('out_file', 'in_file')]),
    ])
else:
    wf.connect([
        (rename_src, sampler, [('out_file', 'in_file')]),
    ])
```